### PR TITLE
feat(k8s-manager): Enrich KubernetesEnterprise classes in kubernetes manager

### PR DESCRIPTION
**Pull request by MaciejTe**
feat(k8s-manager): Enrich KubernetesEnterprise classes in kubernetes manager

Signed-off-by: Maciej Tomczuk <maciej.tomczuk@northern.tech>

### CODE REVIEW

**Comment by MaciejTe** [comment]
@mender-test-bot start pipeline

**Comment by MaciejTe** [comment]
@mender-test-bot start pipeline

File: testutils/infra/container_manager/kubernetes_manager.py [hunk 1]
```diff
@@ -42,22 +63,74 @@ def cmd(self, container_id, docker_cmd, cmd=[]):
...
 
     def get_mender_gateway(self):
         return os.environ.get("GATEWAY_HOSTNAME")
 
 
+class KubernetesEnterpriseSetupWithGateway(KubernetesNamespace):
+    COMPOSE_FILES_PATH = os.path.realpath(
+        os.path.join(os.path.dirname(__file__), "..", "..", "..")
+    )
+    MENDER_GATEWAY_FILES = [
```
Comment (oleorhagen) [inline testutils/infra/container_manager/kubernetes_manager.py:86]: Q: These are duplicated from `docker_compose_manager.py`, right?

Maybe we should split them out in both places, to make this consistent?

File: tests/run.sh [hunk 2]
```diff
@@ -200,8 +200,6 @@ export TENANTADM_STRIPE_API_KEY=$STRIPE_API_KEY
 if [ -n "$K8S" ]; then
     export KUBECONFIG="${HOME}/kubeconfig.${K8S}"
     aws eks update-kubeconfig --region $AWS_DEFAULT_REGION --name $AWS_EKS_CLUSTER_NAME --kubeconfig ${HOME}/kubeconfig.${K8S}
-    kubectl config set-context --current --namespace=$K8S
```
Comment (oleorhagen) [inline tests/run.sh:203]: Q: Not so familiar with k8s. Why remove this?

File: tests/common_setup.py [hunk 3]
```diff
@@ -472,7 +472,7 @@ def create_tenant(env):
     cli = CliTenantadm(containers_namespace=env.name)
     tid = cli.create_org(tname, u.name, u.pwd, plan="os")
 
-    tenant = cli.get_tenant(tid)
+    tenant = cli.get_tenant(tid.strip())
```
Comment (oleorhagen) [inline tests/common_setup.py:475]: Nice!

**Review by oleorhagen — APPROVED** [review]
Left a few questions, I'm not too familiar with k8s. But on the surface looks nice I think :)

File: tests/run.sh (hunk 2 above)
Comment (MaciejTe) [reply tests/run.sh:203]: I added `kubernetes` dependency (which uses k8s API underneath) into `tests/requirements/python-requirements.txt`, which is capable of setting default context and performing all necessary actions on staging k8s cluster.

**Comment by MaciejTe** [comment]
@mender-test-bot start pipeline

File: testutils/infra/container_manager/kubernetes_manager.py (hunk 1 above)
Comment (MaciejTe) [reply testutils/infra/container_manager/kubernetes_manager.py:86]: Thanks for comment! I'll work on this in future PRs; let's see how it behaves on staging environment.

**Comment by MaciejTe** [comment]
@mender-test-bot start pipeline

**Pull request merged**